base_local_planner/src/map_grid.cpp

```diff
@@ -218,17 +218,48 @@ namespace base_local_planner{
     std::vector<geometry_msgs::PoseStamped> adjusted_global_plan;
     adjustPlanResolution(global_plan, adjusted_global_plan, costmap.getResolution());
 
-    // skip global path points until we reach the border of the local map
+    auto dist_squared_to_front = [&adjusted_global_plan](const double x, const double y) {
+      const double dx = x - adjusted_global_plan.front().pose.position.x;
+      const double dy = y - adjusted_global_plan.front().pose.position.y;
+      return dx * dx + dy * dy;
+    };
```

Comment:
one simplification would be to convert this to an approximate index by
`const int path_index_away_from_start = 1 / costmap.getResolution();`